### ISSUE TITLE
[REVIEW]: The ExoTETHyS package: Tools for Exoplanetary Transits around Host Stars

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1834]
**Submitting author:** @gmorello (<a href="http://orcid.org/0000-0002-4262-5661">Giuseppe Morello</a>)
**Repository:** <a href="https://github.com/ucl-exoplanets/ExoTETHyS" target ="_blank">https://github.com/ucl-exoplanets/ExoTETHyS</a>
**Version:** v1.0.1
**Editor:** @arfon
**Reviewer:** @steven-murray, @williamjameshandley
**Archive:** <a href="https://doi.org/10.5281/zenodo.3577740" target="_blank">10.5281/zenodo.3577740</a>

## Status

[![status](https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6/status.svg)](https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6"><img src="https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6/status.svg)](https://joss.theoj.org/papers/4228224c360b4b32075cfe79475005d6)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ucl-exoplanets/ExoTETHyS" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@steven-murray & @williamjameshandley, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @arfon know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ucl-exoplanets/ExoTETHyS">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@gmorello) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ucl-exoplanets/ExoTETHyS">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@gmorello) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @steven-murray, @williamjameshandley it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01834/joss.01834/10.21105.joss.01834.pdf)

**Comment by arfon** [comment]
@steven-murray, @williamjameshandley - One thing I forgot to mention when inviting you both to review is that this JOSS submission/paper is a joint publication with AAS Journals, i.e. this JOSS paper will be published together with another paper in one of the AAS journals. As part of this collaboration, AAS publishing makes a small donation to the running costs of JOSS. 

If this is unacceptable to you, please let me know and I can look for alternative reviewers.

Some related links:

- Blog post describing the collaboration: https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
- Summary of JOSS financial model: https://joss.theoj.org/about#costs

**Comment by steven-murray** [comment]
@arfon it seems I am not able to check off the check-boxes in the original comment?

**Comment by arfon** [comment]
@steven-murray - did you accept the invite here? https://github.com/openjournals/joss-reviews/invitations

**Comment by steven-murray** [comment]
@arfon -- ahhh, OK now I have :-)

**Comment by williamjameshandley** [comment]
@arfon 
> (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

I would consider myself an aspiring completist, but am currently not enough of one in order to disambiguate the above instruction. Could you point me to an example of how to link between a JOSS review and a GitHub issue?

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01834/joss.01834/10.21105.joss.01834.pdf)

**Comment by steven-murray** [comment]
Apart from the code issues, which I have opened in the repository itself, I have the following comments about the paper:

1. While there is a statement of what the software's purpose is, there is no additional context in the opening sentences (i.e. is this an important task for the field? What motivates the construction of this code?)
2. While there is a passing reference to the code being able to outperform other codes, there is no discussion of alternatives to this code, and what motivated the construction of this one, or whether this code is the _only_ way (currently) to use the SAIL and TRIP algorithms/packages.
3. Second sentence: "Now, " can be removed.
4. The last section could be moved to its own paragraph, concerning "how the code is currently being used"
5. It would be good to identify ways the code may be used outside of the current uses.
6. "more details can be found in..." -- perhaps it should be "Scientific details of the algorithms can be found in <>, and technical usage details can be found on GitHub"

**Comment by arfon** [comment]
@williamjameshandley - how are you getting along with your review here?

**Comment by williamjameshandley** [comment]
The repository installs fine on my machine (I'm using the recently released [python 3.8](https://docs.python.org/3/whatsnew/3.8.html), so that's a good sign given that I assume it hasn't necessarily been tested on this version).

- [ ]The [arXiv paper](https://arxiv.org/abs/1908.09599) makes it pretty clear that the intention is to extend this system of package as time goes on, which I approve of. My main requirement for acceptance is that a template for continuous integration is set up which can also be extended as the codebase expands. I've added this as a [GitHub issue](https://github.com/ucl-exoplanets/ExoTETHyS/issues/14), and we can discuss further there.

- [ ]It would also be good to upload this to [PyPi](https://pypi.org/) so that it can be installed without the need to clone the git repo. Also added as an [issue](https://github.com/ucl-exoplanets/ExoTETHyS/issues/15)

Other than this, more minor additions could be:

> 1. SAIL (Stellar Atmosphere Intensity Limb), i.e., a calculator of stellar limb-darkening coeffi-
cients that outperforms the existing software by one order of magnitude in terms of light-curve
model accuracy, i.e., down to <10 parts per million (ppm);
- [ ] This claim could be backed up by a specific example that the user can run

>2. TRIP (Transit Ring-Integrated Profile), which can compute an exact transit light-curve
by direct integration of the occulted stellar flux from the model intensities, without using a
parameterization (limb-darkening law) to approximate the stellar intensity profile.
- [ ] There should be a clear example of this in the README, ideally with a plot to show it working.

**Comment by danielskatz** [comment]
👋 @arfon - How is this review going?

**Comment by steven-murray** [comment]
Currently the author is working on closing the issues that we have made in his repo -- most of them have now been addressed.

**Comment by arfon** [comment]
:wave: @gmorello - happy new year!

Could you give me an update on the status of your updates based on the reviewer feedback?

**Comment by gmorello** [comment]
@steven-murray<https://github.com/steven-murray>

I think I have addressed all your points except the first part of point 2:
> While there is a passing reference to the code being able to outperform other codes, there is no discussion of alternatives to this code

There are many limb-darkening calculators such as SAIL, the discussion of the performances is section 3.1 of the arXiv/AJ paper, but I would avoid to just report a long list of references with a negative statement without detailed explanations. I propose to refer to the AJ paper with the full discussion without additional comments, as it is now. Do you have another idea?

Best regards,
Giuseppe

________________________________

**Comment by steven-murray** [comment]
@gmorello is there an updated version of the paper I can read? 

If you don't want to fully recapitulate the discussion concerning comparison to other codes in this paper, perhaps you could abbreviate it by saying something like "More details, and a comprehensive comparison to other relevant codes, can be found in ....".

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01834/joss.01834/10.21105.joss.01834.pdf)

**Comment by labarba** [comment]
👋  @steven-murray — you can find a new proof above

**Comment by steven-murray** [comment]
That looks good, thank you. I'm happy with this.

**Comment by arfon** [comment]
> That looks good, thank you. I'm happy with this.

Thanks @steven-murray. There are a few checkboxes in the review unchecked still - could you give these a quick look again?

**Comment by steven-murray** [comment]
@arfon ah yes, re-checked them.

**Comment by arfon** [comment]
@gmorello - I've made a couple of minor updates to your paper for readability in https://github.com/ucl-exoplanets/ExoTETHyS/pull/18

Also, do you know that the status of the AAS companion paper is here? Before we can publish this we should add the AAS paper DOI to the YAML header of the `paper.md`.

**Comment by arfon** [comment]
/ cc @crawfordsm for visibility on this submission that has now completed review and is accepted into JOSS.

**Comment by arfon** [comment]
@gmorello - could you please merge this PR too? https://github.com/ucl-exoplanets/ExoTETHyS/pull/19 . It adds the AAS information to the JOSS paper.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.3577740 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3577740" target="_blank">10.5281/zenodo.3577740</a> is the archive.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1111/j.1365-2966.2010.17094.x is OK
- 10.1093/mnras/stz2958 is OK
- 10.1038/s41550-019-0878-9 is OK

MISSING DOIs

- https://doi.org/10.3847/1538-3881/ab63dc may be missing for title: The ExoTETHyS package: Tools for Exoplanetary Transits around Host Stars

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1297

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1297, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1227184499328536577) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1298
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01834](https://doi.org/10.21105/joss.01834)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@steven-murray, @williamjameshandley - many thanks for your reviews here ✨ 

@gmorello - your paper is now published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01834/status.svg)](https://doi.org/10.21105/joss.01834)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01834">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01834/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01834/status.svg
   :target: https://doi.org/10.21105/joss.01834
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01834/status.svg)](https://doi.org/10.21105/joss.01834)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
